### PR TITLE
Add argument for balena registry

### PR DESCRIPTION
**Pull request by saintaardvark**
This PR fixes https://github.com/balena-io/device-diagnostics/issues/249 by adding a `--balenaos-registry` argument, as is done for the `upgrade-2.x.sh` host update script.

This script sources `/usr/sbin/resin-vars`.  Newer versions of this script do not set `REGISTRY_ENDPOINT`, which is why we're passing in the endpoint via the new argument.  To avoid having this argument overwritten by older versions of resin-vars, we save the CLI arguments before sourcing `resin-vars`, then restore them afterward.

# Testing

## Fixing the problem on balenaOS 2.58.3

First off, running with no arguments demonstrates the original problem:

```
# /tmp/checks.sh  | jq '.checks[] | select(.name == "check_networking" )'

{
  "diagnose_version": "4.20.13",
  "checks": [
[snip]

    {
      "name": "check_networking",
      "success": false,
      "status": "Some networking issues detected: \ntest_balena_registry: Could not communicate with  for authentication"
    },
```

Second, running with a bogus registry shows that it tries to communicate with it:

```
# /tmp/checks.sh --balenaos-registry doesnotexist.example  | jq '.checks[] | select(.name ==
 "check_networking" )'
{
  "name": "check_networking",
  "success": false,
  "status": "Some networking issues detected: \ntest_balena_registry: Could not communicate with doesnotexist.example for authentication"
}
```

Finally, running with the correct argument (as will be done in the proxy) shows that it works:

```
# /tmp/checks.sh --balenaos-registry registry2.balena-staging.com  | jq '.checks[] | select(.name == "check_networking" )'
{
  "name": "check_networking",
  "success": true,
  "status": "No networking issues detected"
}
```

## Demonstrating no regression with balenaOS 2.48

First off, running with no arguments demonstrates that everything works as before:

```
# /tmp/checks.sh  | jq '.checks[] | select(.name == "check_networking" )'
{
  "name": "check_networking",
  "success": true,
  "status": "No networking issues detected"
}
```

Second, running with a bogus registry shows that it tries to communicate with it (just as with 2.58.3):

```
# /tmp/checks.sh --balenaos-registry doesnotexist.example  | jq '.checks[] | select(.name == "check_networking" )'
{
  "name": "check_networking",
  "success": false,
  "status": "Some networking issues detected: \ntest_balena_registry: Could not communicate with doesnotexist.example for authentication"
}
```

Finally, specifying the correct registry works as expected:

```
# /tmp/checks.sh --balenaos-registry registry2.balena-staging.com  | jq '.checks[] | select(.name == "check_networking" )'
{
  "name": "check_networking",
  "success": true,
  "status": "No networking issues detected"
}
```

# What about the proxy?

Deployment in the proxy will need to be paired with a separate PR to add this missing argument; that will be added shortly.

Connected-to: https://github.com/balena-io/device-diagnostics/issues/249
Change-type: patch
Signed-off-by: Hugh Brown <hugh@balena.io>

### CODE REVIEW

File: scripts/checks.sh [hunk 1]
```diff
@@ -563,5 +589,25 @@ function run_checks()
 	| jq -s 'add | {checks:.}'
 }
 
+# Parse arguments
+while [[ $# -gt 0 ]]; do
+    arg="$1"
+
+    case $arg in
+	-h|--help)
```
Comment (xginn8) [inline scripts/checks.sh:597]: indentation looks funky here

File: scripts/checks.sh [hunk 2]
```diff
@@ -563,5 +589,25 @@ function run_checks()
...
+	    help
+	    exit 0
+	    ;;
+        --resinos-registry | --balenaos-registry)
```
Comment (xginn8) [inline scripts/checks.sh:601]: let's drop `--resinos` support unless there's a good reason to keep it around (i don't believe there is)

File: scripts/checks.sh [hunk 3]
```diff
@@ -3,8 +3,20 @@ DIAGNOSE_VERSION=4.20.13
 # Don't run anything before this source as it sets PATH here
 # shellcheck disable=SC1091
 source /etc/profile
+
+# Newer OS versions do not set REGISTRY_ENDPOINT via
```
Comment (xginn8) [inline scripts/checks.sh:7]: generally speaking we should just trust the cloud and ignore whatever we get on device, as that is the source of truth going forward.

Comment (saintaardvark) [reply scripts/checks.sh:7]: Good point -- I'll add that to the comment, as I think it definitely helps add context here.

**Review by xginn8 — APPROVED** [review]

**Pull request merged**